server/cdn.go

```diff
@@ -47,14 +50,34 @@ func (c *cdn) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	item, ok := c.cache[req.URL.Path] // TODO: respect cache timeouts
 	c.mu.RUnlock()
 
+	// if ok { // We have the data!!!
+	// 	item.Send(w)
+	// } else if req.Header.Get(cdnHeader) != "" {
+	// 	log.Print(c.me + " couldn't find response for neighbor")
+	// 	http.NotFound(w, req) // Request was from other CDN server, don't ask others or origin
+	// } else if item, ok = c.checkNeighbors(req.URL.Path); ok {
+	// 	log.Print(c.me + " Check neighbors called from CDNServerHTTP")
+	//
+	// 	item.Send(w) // Found request on neighbor, sending response
+	// } else {
+	// 	log.Print(c.me + " Headed to origin")
+	// 	c.rp.ServeHTTP(w, req) // Couldn't find it anywhere, sending to origin
+	// }
+
 	if ok { // We have the data!!!
 		item.Send(w)
-	} else if req.Header.Get(cdnHeader) != "" {
-		log.Print(c.me + " couldn't find response for neighbor")
-		http.NotFound(w, req) // Request was from other CDN server, don't ask others or origin
-	} else if item, ok = c.checkNeighbors(req.URL.Path); ok {
-		item.Send(w) // Found request on neighbor, sending response
+		log.Print(c.me + " owns data and sending back")
 	} else {
-		c.rp.ServeHTTP(w, req) // Couldn't find it anywhere, sending to origin
+		serverName := c.dht.Who(req.URL.Path)
```

Comment:
this could be part of the else

```go
} else if c.dnt.Who(req.URL.Path) == c.me {
  c.rp.ServeHTTP(w, req)
} else {
  result, _ := c.DHTFetch(req.URL.Path, serverName)
  result.Send(w)
}
```